plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Check/Scope.hs

```diff
@@ -45,3 +45,4 @@ checkScope = go 0
         Force _ t     -> go lvl t
         Delay _ t     -> go lvl t
         _             -> pure ()
+{-# INLINE checkScope #-}
```

Comment:
Comment, please! Something like 

Inlining this function makes a big difference, since it will usually be called in a context where all the type variables are known. That then means that GHC can optimize `go` locally in a completely monomorphic setting, which helps a lot.